src/config-import.js

```diff
@@ -0,0 +1,271 @@
+import yauzl from 'yauzl-promise'
+import { validate, valueSchemas } from '@mapeo/schema'
+import { json, buffer } from 'node:stream/consumers'
+import path from 'node:path'
+
+// Throw error if a zipfile contains more than 10,000 entries
+const MAX_ENTRIES = 10_000
+
+/**
+ * @typedef {Omit<yauzl.Entry, 'fileName'> & { filename: string }} Entry
+ * `@types/yauzl-promise` has the wrong name for this key. We should
+ * eventually submit this patch upstream.
+ */
```

Comment:
It looks like you're now getting type errors because this is insufficient. I think we can fix it by updating `types/yauzl-promise.d.ts`:

```diff
diff --git a/types/yauzl-promise.d.ts b/types/yauzl-promise.d.ts
index 1faad07..acf6a23 100644
--- a/types/yauzl-promise.d.ts
+++ b/types/yauzl-promise.d.ts
@@ -3,4 +3,10 @@ declare module 'yauzl-promise' {
   export interface ZipFile extends yauzl.ZipFile {
     [Symbol.asyncIterator]()
   }
+
+  export interface Entry extends yauzl.Entry {
+    // `@types/yauzl-promise` has the wrong name for this key.
+    // We should eventually submit this patch upstream.
+    filename: string;
+  }
 }
```

We can then import the Entry type which should fix things:

```suggestion
/** @typedef {yauzl.Entry} Entry */
```